nixos/modules/services/networking/nomad.nix

```diff
@@ -139,9 +139,18 @@ in
         {
           DynamicUser = cfg.dropPrivileges;
           ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
-          ExecStart = "${cfg.package}/bin/nomad agent -config=/etc/nomad.json" +
+          ExecStart =
+            let
+              pluginDir = pkgs.runCommandNoCC "plugin"
+                { preferLocalBuild = true; }
+                ''
+                  mkdir -p $out/bin/
+                  ${lib.concatStringsSep "\n" (map (p: "ln -s ${p}/bin/* $out/bin") cfg.extraPlugins)}
```

Comment:
Did you try running this code?

If I'm reading this correctly you'll generate a line like this for every plugin in `cfg.extraPlugins`:

```sh
ln -s foo_plugin/bin/* $out/bin
```

Since you're creating `$out/bin` with `mkdir` above, you can't then symlink `$out/bin` to something else.

Additionally, even if you weren't creating `$out/bin` above, what does it mean to create a symlink to a glob? I don't think that's a valid use of `ln`.

Reply:
![image](https://user-images.githubusercontent.com/21156405/143782526-7c3085b7-c641-4652-9f88-fab6ae3c3c26.png)
Yeh. I did, it works well for me, and tested with both podman and nix.

Reply:
> Additionally, even if you weren't creating $out/bin above, what does it mean to create a symlink to a glob? I don't think that's a valid use of ln.

the plugin-dir flag does not support several flags in search dir. So, we need a dir which included all of bin execs in there. Also, I think the `buildEnv` better than `runCommand` creating bin link.